packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -1253,6 +1253,7 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
       return MatrixUtils.transformRect(transform, rect ?? descendant.paintBounds);
     }
 
+    assert(targetOffset.offset.isFinite);
```

Comment:
Is this assertion necessary?

Reply:
I ask because I wonder what this scenario is, and if we can catch it sooner in a place that would provide better context to the user in an error message to help debug. :)

Reply:
Unfortunately, there is no better place to assert, and useful debugging messages cannot be provided here.

Reply:
Should the SliverMainAxisGroup assert its children have a finite extent if there are more children expected to follow in the group? That seems a better place to assert from. The issue would be coming from the group, not the whole viewport, which would probably be harder to debug.